src/scenes/MainScene.ts

```diff
@@ -24,21 +25,17 @@ export default class MainScene extends Phaser.Scene {
     );
 
     this.ground = new Ground(this, hills);
-    this.ball = new Ball(this);
-    this.goal = new Goal(this);
+    this.ball = new Ball(this).element;
+    this.goal = new Goal(this).element;
 
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    this.ball.setOnCollideWith(this.goal, () => {
+    const goalBody = this.goal?.body as BodyType;
```

Comment:
I dont like this but the typings in Phaser3 are neither strict enough nor generic to allow for a better solution

Reply:
yoooo thank you so much